internal/core/server/workspace.go

```diff
@@ -433,16 +433,27 @@ func (ws *Workspace) getEnvironment(ctx context.Context) (map[string]string, err
 
 func (ws *Workspace) CreateComponent(ctx context.Context, input *api.CreateComponentInput) (*api.CreateComponentOutput, error) {
 	id := gensym.RandomBase32()
-	err := ws.createComponent(ctx, manifest.Component{
-		Name: input.Name,
-		Type: input.Type,
-		Spec: input.Spec,
-	}, id)
-	if err != nil {
-		return nil, err
-	}
+	job := ws.TaskTracker.StartTask(ctx, "creating "+input.Name)
+
+	go func() {
+		defer job.Finish()
+		err := ws.createComponent(ctx, manifest.Component{
```

Comment:
Hm, I would not have thought to do it this way - it does work, but it introduces a new problem. I was expecting that `createComponent` would get split in to synchronous and asynchronous parts, so that errors could be returned synchronously for spec changes where those failures can be detected up front. Primarily: name conflicts.

Reply:
I forgot about name conflicts. My assumption was that in practice creating the spec would not fail, and this would be "good enough" until the component model gets reworked, but I don't think that's the case. I'll go ahead and make this change.